### PR TITLE
feat: node 18 support

### PR DESCRIPTION
**Pull request by nataliepina**
Closes D2IQ-96159

# Description
Now that we've updated to Storybook 7, theoretically, this should be good to go now.

To break down some of the changes further. This should mimic what we have in kommander-ui where we are able to set up a single file to reference the node version we want to use. It will be easier to update the node version by only having to update this one location. 

I've updated the actions to use a matrix testing strategy.  This configuration creates three parallel jobs, one for each value specified in the node-version array. Each job runs with a different version of Node.js: 16 and 18. The matrix strategy is useful when you want to run the same job with different configurations or environments. In this case, it allows the action to be tested with multiple Node.js versions to ensure compatibility and verify the behavior across different runtime environments.

## Testing
We'll probably need to merge this. 

## Checklist

- [x] This PR is associated with a JIRA and it is mentioned in the commit message footer ("Closes …")
- [ ] Significant changes have been tested downstream to avoid breaking changes
- [ ] This PR contains breaking changes and states in the commit message body ("BREAKING CHANGE: …")
- [ ] I have reviewed the changes and provided detail to the sections above

### CODE REVIEW

**Comment by nataliepina** [comment]
I think this will fail until we fix the gh actions using build-storybook, but this can still be reviewed. It may block this from merge. We have a task related to fixing this in the sprint.

**Comment by nataliepina** [comment]
> I think this will fail until we fix the gh actions using build-storybook, but this can still be reviewed @d2iq/ui-team. It may block this from merge. We have a task related to fixing this in the sprint.

File: .node-version [hunk 1]
```diff
@@ -0,0 +1 @@
+18.15.0
```
Comment (mstruebing) [inline .node-version:1]: Should we use 18.16.0?

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM node:16.13.0-slim
+FROM node:18.15.0-slim
```
Comment (mstruebing) [inline Dockerfile:1]: Do we even use this `Dockerfile` anywhere? Can we remove this?

File: .node-version (hunk 1 above)
Comment (nataliepina) [reply .node-version:1]: I wanted to match to what we have in kommander-ui. Should we update it there if I move this to the latest?

File: Dockerfile (hunk 2 above)
Comment (nataliepina) [reply Dockerfile:1]: 🤔  Let me check and i'll remove it if not.

Comment (nataliepina) [reply Dockerfile:1]: It's used by Uffizzi.

File: .node-version (hunk 1 above)
Comment (nataliepina) [reply .node-version]: Updated this to 18.16.0

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -5,11 +5,14 @@ on:
...
+    strategy:
+      matrix:
+        node-version: [16, 18]
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - uses: actions/setup-node@v2
         with:
-          node-version: "16"
+          node-version: ${{ matrix.node-version }}
```
Comment (mstruebing) [inline .github/workflows/release.yml:15]: Isn't one node version enough for release?
I would suggest using the one from `.node-version`

```suggestion
          node-version-file: ".node-version"
```

And removing line 8-10.

Comment (nataliepina) [reply .github/workflows/release.yml:15]: @clintonmedbery suggested the matrix for testing. Clint do you have any thoughts on the benefits of the matrix vs a single version?

Comment (mstruebing) [reply .github/workflows/release.yml:15]: This is the `release` workflow, not the testing workflow :)

Comment (clintonmedbery) [reply .github/workflows/release.yml]: Yeah I like the idea of testing on multiple versions (if that is what we want to do) but I think the release only needs one version.

**Review by clintonmedbery — APPROVED** [review]
Nice work!

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -6,11 +6,14 @@ on:
 jobs:
   ci:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [16, 18]
```
Comment (mstruebing) [inline .github/workflows/ci.yml:11]: ❓ Why do we want to run the CI pipeline with node 16 and 18 when we use node 18 in development and we do not even publish a node package but a react component library?

Comment (nataliepina) [reply .github/workflows/ci.yml:11]: I'm just going to remove this. I'd like to get this merged in before i'm off.

**Review by mstruebing — APPROVED** [review]

**Pull request merged**